micropython/test_micropython.c

```diff
@@ -0,0 +1,126 @@
+#ifdef __STDC_ALLOC_LIB__
+#define __STDC_WANT_LIB_EXT2__ 1
+#endif
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <string.h>
+
+#define MICROPYTHON   "/bin/micropython "
+#define PATH_TO_TESTS "/usr/test/micropython/"
+
+#define CMDLINE_OPTION_IDEN "# cmdline: "
+
+void exit_with_error(char *msg)
+{
+	fprintf(stderr, "Error: %s\n", msg);
+	exit(1);
+}
+
+char *cat(const char *str1, const char *str2)
+{
+	int length = strlen(str1) + strlen(str2) + 1;
+	char *result = (char *)malloc(sizeof(char) * length);
+	if (!result) {
+		return NULL;
+	}
+
+	result[0] = '\0';
+	strcat(result, str1);
+	strcat(result, str2);
+
+	return result;
+}
+
+char *check_for_options(const char *path)
+{
+	FILE *f;
+	char *line = NULL;
+	size_t len, result_len, opt_iden_len = strlen(CMDLINE_OPTION_IDEN);
+	ssize_t read;
+	char *result;
+
+	f = fopen(path, "r");
+	if (!f)
+		exit_with_error("There was a problem with opening file with a script");
+
+	if (getline(&line, &len, f) < 0)
+		exit_with_error("There was a problem with reading script to execute");
+
+	len = strlen(line) + 1;
+
+	if (len > opt_iden_len &&
+		strncmp(line, CMDLINE_OPTION_IDEN, opt_iden_len) == 0) {
+		result_len = strlen(line) - opt_iden_len;
+		result = (char *)malloc(sizeof(char) * result_len);
+		if (!result)
+			exit_with_error("There is no enough memory to run this test");
+
+		strcpy(result, &line[opt_iden_len]);
+
+		/* Changing the last character from '\n' to ' ' */
+		result[strlen(result) - 1] = ' ';
+	}
+	else
+		result = NULL;
```

Comment:
We can skip this `else`, if `result` would be declared with NULL value, and then simply returned as it is now